### PR TITLE
Add docs for the configuration input detection opt-out flags

### PR DESCRIPTION
**Pull request by h0tk3y**
Adds the missing docs for:
* https://github.com/gradle/gradle/pull/25544
* https://github.com/gradle/gradle/pull/25622

### CODE REVIEW

**Comment by ljacomet** [comment]
Given the discussions around file-system-checks, I believe adding something to the release notes is needed.

**Comment by h0tk3y** [comment]
@ljacomet 
> Given the discussions around file-system-checks, I believe adding something to the release notes is needed.

Yes, we are going to mention the changes in the release notes, too.

**Review by bamboo — APPROVED** [review]
Great job 👍

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 1]
```diff
@@ -567,6 +567,50 @@ Let your team know they can opt-in by, for example, enabling the configuration c
 Later on, when more workflows are working, you can flip this around.
 Enable the configuration cache by default, configure CI to disable it, and if required communicate the unsupported workflow(s) for which the configuration cache needs to be disabled.
 
+=== Adopting changes in the configuration cache behavior
+
+Gradle releases bring enhancements in the configuration cache, making it detect more cases of configuration logic interacting with the environment.
+Those changes eventually improve the correctness of the cache by eliminating potential false cache hits.
+On the other hand, they impose stricter rules that plugins and build logic need to follow in order to be cached as often as possible.
+
+Some of those configuration inputs may be considered "benign" if their results do not actually affect the configured tasks.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:576]:
```suggestion
Some of those configuration inputs may be considered "benign" if their results do not affect the configured tasks.
```

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 2]
```diff
@@ -567,6 +567,50 @@ Let your team know they can opt-in by, for example, enabling the configuration c
...
+Having new configuration misses because of them may be undesirable for the build users, and the suggested strategy for eliminating them is:
+
+* Identify the configuration inputs causing invalidation of the configuration cache with the help of the <<configuration_cache#config_cache:troubleshooting, configuration cache report>>.
+** Fix undeclared configuration inputs accessed by the build logic of the project.
+** Report issues caused by third-party plugins to the plugin maintainers, and update the plugins once they get fixed.
+
+* For some kinds of configuration inputs, it is possible to use the opt-out options that make Gradle fall back to the earlier behavior, omitting the inputs from detection.
+This temporary workaround is aimed to mitigate performance issues coming from out-of-date plugins.
+
+It is possible to temporarily opt-out of configuration input detection in the following cases:
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:586]:
```suggestion
It is possible to temporarily opt out of configuration input detection in the following cases:
```

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 3]
```diff
@@ -567,6 +567,50 @@ Let your team know they can opt-in by, for example, enabling the configuration c
...
+----
+
+* Before Gradle 8.3, some undeclared configuration inputs that were never used in the configuration logic could still be read when the task graph was serialized by the configuration cache.
+However, their changes would not invalidate the configuration cache afterward.
+Starting with Gradle 8.3, such undeclared configuration inputs are correctly tracked.
++
+To temporarily revert to the earlier behavior, set the Gradle property `org.gradle.configuration-cache.inputs.unsafe.ignore-in-serialization` to `true`.
+
+One should ignore configuration inputs sparingly, and only if those do not affect the tasks produced by the configuration logic.
+The support for these options will be removed in the future releases.
```
Comment (lkasso) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:611]:
```suggestion
The support for these options will be removed in future releases.
```

**Review by lkasso — APPROVED** [review]

**Comment by h0tk3y** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 4]
```diff
@@ -567,6 +567,50 @@ Let your team know they can opt-in by, for example, enabling the configuration c
...
+----
+org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks=\
+    ~/.third-party-plugin/*.lock;\
+    ../../externalOutputDirectory/**;\
+    build/analytics.json
+----
+
+* Before Gradle 8.3, some undeclared configuration inputs that were never used in the configuration logic could still be read when the task graph was serialized by the configuration cache.
+However, their changes would not invalidate the configuration cache afterward.
+Starting with Gradle 8.3, such undeclared configuration inputs are correctly tracked.
```
Comment (mlopatkin) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:606]: Shouldn't it be `Gradle 8.4` here, as `master` is already at 8.4?

Comment (h0tk3y) [reply subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:606]: Yes, it should, I was going to update it after finalizing the actual 8.3 docs.

**Pull request merged**